### PR TITLE
Move queen

### PR DESCRIPTION
**Pull request by emathew80**

### CODE REVIEW

File: app/models/queen.rb [hunk 1]
```diff
@@ -1,2 +1,17 @@
 class Queen < Piece
+  def valid_move?(destination_x, destination_y)
+    return false unless on_board?(destination_x, destination_y)
```
Comment (sylyu) [inline app/models/queen.rb]: we don't need to check this anymore.  Onboard is checked in obstructed? now.

File: app/models/queen.rb [hunk 2]
```diff
@@ -1,2 +1,17 @@
...
+    return false if obstructed?(destination_x, destination_y)
+
+    x_diff = (x_coordinate - destination_x).abs
+    y_diff = (y_coordinate - destination_y).abs
+
+    return true if x_diff == y_diff
+
+    return true if x_diff.zero?
```
Comment (sylyu) [inline app/models/queen.rb]: queen also slides vertically and horizontally.

File: app/models/queen.rb (hunk 1 above)
Comment (emathew80) [inline app/models/queen.rb]: Oh okay!! Thanks

File: app/models/queen.rb (hunk 2 above)
Comment (sylyu) [inline app/models/queen.rb]: horizontal and vertical move code is missing.

**Pull request merged**